packages/hardhat/contracts/FundingRound.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title FundingRound
+ * @dev This contract allows investors to deposit USDC until a capital raised goal is reached.
+ * Optionally, an investor whitelist can be specified at construction.
+ */
+contract FundingRound is Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @notice Address of the USDC token contract.
+    IERC20 public immutable usdc;
+    /// @notice Capital raised goal in USDC.
+    uint256 public immutable goal;
+    /// @notice Indicates if a whitelist is used.
+    bool public immutable whitelistEnabled;
+    /// @notice Total amount of USDC raised.
+    uint256 public totalRaised;
+    /// @notice Mapping to track individual contributions.
+    mapping(address => uint256) public contributions;
+    /// @notice Mapping for whitelist of approved investors.
+    mapping(address => bool) public whitelist;
```

Comment:
this should be address->uint256 because the founders chooses the amount that the investor can invest in case of private rounds

Reply:
Ook, but the wl wont be opcional anymore